src/com/vivian/java/leetcode/Q13_RomantoInteger.java

```diff
@@ -1,10 +1,62 @@
 package com.vivian.java.leetcode;
 
 /**
- * Given a roman numeral, convert it to an integer.
+ * Given a roman string, convert it to an integer.
  * 
  * Input is guaranteed to be within the range from 1 to 3999.
  */
 public class Q13_RomantoInteger {
+    // 右加左减
+    public int romanToInteger(String str) {
+        int current;
+        int pervious = 0;
+        int sum = 0;
+        if (str.isEmpty() || str == null) {
+            return -1;
+        }
+        for (int i = str.length() - 1; i >= 0; i--) {
+            current = romanNumber(str.charAt(i));
+            if (i == str.length() - 1) {
+                sum = sum + current;
```

Comment:
Actually you don't have to treat the case specially, as when i == str.length() - 1, previous is 0, and is guaranteed to be less than current, thus, line 23 will be executed.